metricshub-hardware/src/main/java/org/sentrysoftware/metricshub/hardware/threshold/FanMetricNormalizer.java

```diff
@@ -21,25 +21,148 @@
+import org.sentrysoftware.metricshub.engine.telemetry.metric.NumberMetric;
 
 /**
- * TODO: Complete the Javadoc for this Class.
+ * The FanMetricNormalizer class is responsible for normalizing FAN metrics.
+ * It extends the AbstractMetricNormalizer class to provide specific
+ * normalization logic for FAN monitor hardware metrics.
  */
 public class FanMetricNormalizer extends AbstractMetricNormalizer {
 
+	private static final Double DEFAULT_LOW_DEGRADED_VALUE_SPEED_LIMIT_METRIC = 500.0;
+	private static final Double DEFAULT_LOW_CRITICAL_VALUE_SPEED_LIMIT_METRIC = 0.0;
+	private static final Double DEFAULT_LOW_DEGRADED_VALUE_SPEED_RATIO_LIMIT_METRIC = 0.05;
+	private static final Double DEFAULT_LOW_CRITICAL_VALUE_SPEED_RATIO_LIMIT_METRIC = 0.0;
+
 	/**
 	 * Constructs a new instance with the specified strategy time.
 	 * @param strategyTime The strategy time in milliseconds
-	 * @param hostname     The hostname of the monitor
+	 * @param hostname The hostname of the monitor
 	 */
 	public FanMetricNormalizer(long strategyTime, String hostname) {
 		super(strategyTime, hostname);
 	}
 
 	/**
-	 * TODO: Complete the Javadoc for this method.
+	 * Normalizes fan speed and speed ratio limit metrics
+	 * @param monitor A given {@link Monitor}
 	 */
 	@Override
-	public void normalize(Monitor monitor) {}
+	public void normalize(Monitor monitor) {
+		normalizeSpeedLimitMetric(
+			monitor,
+			"hw.fan.speed",
+			DEFAULT_LOW_DEGRADED_VALUE_SPEED_LIMIT_METRIC,
+			DEFAULT_LOW_CRITICAL_VALUE_SPEED_LIMIT_METRIC
+		);
+		normalizeSpeedLimitMetric(
+			monitor,
+			"hw.fan.speed_ratio",
+			DEFAULT_LOW_DEGRADED_VALUE_SPEED_RATIO_LIMIT_METRIC,
+			DEFAULT_LOW_CRITICAL_VALUE_SPEED_RATIO_LIMIT_METRIC
+		);
+	}
+
+	/**
+	 * Normalizes the speed limit metrics.
+	 * @param monitor The monitor to normalize
+	 * @param metricNamePrefix The prefix of the metric name.
+	 * @param defaultLowDegradedValueMetric The default value for the low degraded limit metric.
+	 * @param defaultLowCriticalValueMetric The default value for the low critical limit metric.
+	 */
+	public void normalizeSpeedLimitMetric(
+		final Monitor monitor,
+		final String metricNamePrefix,
+		final Double defaultLowDegradedValueMetric,
+		final Double defaultLowCriticalValueMetric
+	) {
+		if (!isMetricCollected(monitor, metricNamePrefix)) {
+			return;
+		}
+
+		// Get the low degraded metric
+		final Optional<NumberMetric> maybeLowDegradedMetric = findMetricByNamePrefixAndAttributes(
+			monitor,
+			String.format("%s.limit", metricNamePrefix),
+			Map.of("limit_type", "low.degraded")
+		);
+
+		// Get the low critical metric
+		final Optional<NumberMetric> maybeLowCriticalMetric = findMetricByNamePrefixAndAttributes(
+			monitor,
+			String.format("%s.limit", metricNamePrefix),
+			Map.of("limit_type", "low.critical")
+		);
+
+		// If neither low degraded nor low critical metrics are available, create both
+		if (maybeLowDegradedMetric.isEmpty() && maybeLowCriticalMetric.isEmpty()) {

```

Comment:
We can simplify this block as follows:
```java


    if (maybeLowDegradedMetric.isEmpty() && maybeLowCriticalMetric.isEmpty()) {
        final MetricFactory metricFactory = new MetricFactory(hostname);
        // Create both metrics if neither are present
        collectMetric(monitor, metricNamePrefix, metricFactory, "low.degraded", defaultLowDegradedValueMetric);
        collectMetric(monitor, metricNamePrefix, metricFactory, "low.critical", defaultLowCriticalValueMetric);
    } else if (maybeLowDegradedMetric.isPresent() && maybeLowCriticalMetric.isPresent()) {
        // Adjust values if both metrics are present
        adjustMetricsIfNecessary(maybeLowDegradedMetric.get(), maybeLowCriticalMetric.get());
    } else if (maybeLowDegradedMetric.isEmpty()) {
        // Create degraded metric if only critical is present
        collectMetric(
          monitor,
          metricNamePrefix,
          new MetricFactory(hostname),
          "low.degraded",
          maybeLowCriticalMetric.get().getValue() * 1.1
      );
    } else {
        // Create critical metric if only degraded is present
        collectMetric(
          monitor,
          metricNamePrefix,
          new MetricFactory(hostname),
          "low.critical",
           maybeLowDegradedMetric.get().getValue() * 0.9
      );
    }
```

Reply:
@NassimBtk @SafaeAJ I think there is something missing here: the collectMetric method that we use  is not retrieving the attributes from the present metric.

Reply:
@NassimBtk @SafaeAJ, I suggest 

/**
	 * Collects a metric with the specified limit type and value. It also copies other metric attributes if they are not null.
	 *
	 * @param monitor          The monitor to collect the metric
	 * @param metricNamePrefix The prefix of the metric name
	 * @param limitType        The limit type of the metric (e.g., "low.degraded", "low.critical")
	 * @param value The value of the metric
	 * @param otherMetricAttributes The attributes of another metric to be added after the current metric creation
	 */
	protected void collectMetric(
		final Monitor monitor,
		final String metricNamePrefix,
		final String limitType,
		final Double value,
		final Map<String, String> otherMetricAttributes
	) {
		final MetricFactory metricFactory = new MetricFactory(hostname);
		final  NumberMetric metric = metricFactory.collectNumberMetric(
			monitor,
			String.format("%s{limit_type=\"%s\"}", metricNamePrefix, limitType),
			value,
			strategyTime
		);
		// If we need to store any attributes from another limit metric, we exclude the limit_type attribute and we copy all the other ones
	    if (otherMetricAttributes != null && !otherMetricAttributes.isEmpty()) {
			metric.getAttributes().putAll(otherMetricAttributes.entrySet().stream()
					.filter(entry -> !entry.getKey().equals("limit_type"))
					.collect(Collectors.toMap(
							Map.Entry::getKey,  // Key mapping function
							Map.Entry::getValue // Value mapping function
					)));
		}
	}

Reply:
Yes, so that we homogenize the attributes.